app/experimenter/nimbus-ui/src/components/FormExperimentOverviewPartial/index.tsx

```diff
@@ -33,31 +46,56 @@ const FormExperimentOverviewPartial = ({
   const nameValidated = (
     name: string,
     // This could be ValidationRules from react-hook-form/validator, but
-    // register() has several signatures and only this one produces a ref 
-    validateRule: Parameters<typeof register>[1] = { required: "Required" },
+    // register() has several signatures and only this one produces a ref
+    validateRule: Parameters<typeof register>[1] = {
+      required: "This field may not be blank.",
+    },
   ) => ({
     name,
-    isInvalid: touched[name] && errors[name],
-    isValid: touched[name] && !errors[name],
+    isInvalid: !!submitErrors[name] || (touched[name] && errors[name]),
+    isValid: !submitErrors[name] && touched[name] && !errors[name],
     ref: register(validateRule),
   });
 
+  const FormErrors = ({ name }: { name: string }) => (
+    <>
+      {!errors[name] ? (
+        ""
+      ) : (
```

Comment:
Ah, yup, that would be simpler. I see what I did... at one point this only had one of the two error messages, and React didn't like that `{condition && </>}` returned undefined for the overall component. But now that there's two things here, it's at least an empty component rather than undefined.